phoenix-core/src/main/java/org/apache/phoenix/compile/SubselectRewriter.java

```diff
@@ -188,6 +190,13 @@ private SelectStatement flatten(SelectStatement select, SelectStatement subselec
             }
         }
         
+        OffsetNode offset = select.getOffset();
+        if (offsetRewrite != null || (limitRewrite != null & offset != null)) {
```

Comment:
Based on this not being caught by one of your new unit tests, I'd say that there's likely a hole in the test coverage. Would you mind adding a unit test that would have caught this?